LDAP/ldap2s3/ldap2s3.py

```diff
@@ -112,21 +126,89 @@ def normalize_pgp(self, pgplist):
 
         return ret
 
+    def validate_user(self, user_profile):
+        """
+        Validate a user profile against CIS profile
+        Returns True if the profile validate, False if not.
+        @user_profile: dict
+        """
+        schema_file = self.cis_config.schema
+        try:
+            with open(schema_file) as fd:
+                schema = json.load(fd)
+        except FileNotFoundError:
+            return False
+
+        try:
+            jsonschema.validate(user_profile, schema)
+            return True
+        except jsonschema.exceptions.ValidationError:
+            return False
+
     def user(self, entry):
         """
         Finds canonical LDAP data for that user
         """
-        user = DotDict(dict())
-        attrs = entry.get('raw_attributes')
-        user.mail = self.gfe(attrs, 'mail')
-        user.dn = self.gfe(entry, 'raw_dn')
-        if not user.mail or not user.dn:
-            logger.warning('Invalid user specification dn: {} mail: {}'.format(user.dn, user.mail))
-
-        user.sshPublicKey = self.normalize_ssh(attrs.get('sshPublicKey'))
-        user.pgpFingerprint = self.normalize_pgp(attrs.get('pgpFingerprint'))
+        user = DotDict(json.loads(self.userprofile_json))
 
-        return user
+        # Add LDAP attributes
+        attrs = entry.get('raw_attributes')
+        dn = self.gfe(entry, 'raw_dn')
+
+        # Insert LDAP email as primary email
+        user.primaryEmail = self.gfe(attrs, 'mail')
+        email = {
+                  'value': user.primaryEmail,
+                  'verified': True,
+                  'primary': True,
+                  'name': 'LDAP-imported Email'
```

Comment:
@andrewkrug since you use something similar in the current LDAP publisher  (the name string) i could actually copy your exact string, if this helps. (well, especially for ssh/pgp keys below i guess)
lemme know

Reply:
It will get overwritten by whatever is here.  So NBD.  I just think it's important to let the user know where the value came from.